src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -709,6 +712,10 @@ QgsIdentifyResultsFeatureItem *QgsIdentifyResultsDialog::createFeatureItem( QgsV
       defVal = vlayer->dataProvider()->defaultValueClause( fields.fieldOriginIndex( i ) );
 
     const QString originalValue = defVal == attrs.at( i ) ? defVal : fields.at( i ).displayString( attrs.at( i ) );
+    if ( originalValue.size() == 0 && QgsSettings().value( QStringLiteral( "/Map/hideNullValues" ), false ).toBool() )
```

Comment:
Checking for an empty string != a null value. This should instead be something like `if ( attrs.at( i ).isNull() )`